modules/gojira-core/src/main/java/com/flipkart/gojira/core/aspects/InvocationJoinPointClosure.java

```diff
@@ -40,7 +40,12 @@ public AccessibleObject getStaticPart() {
         ret = clazz.getMethod(codeSignature.getName(), codeSignature.getParameterTypes());
       }
     } catch (NoSuchMethodException e) {
-      throw new UnsupportedOperationException("Can't find member " + codeSignature.toLongString());
+      try {
+        ret = clazz.getDeclaredMethod(codeSignature.getName(), codeSignature.getParameterTypes());
+      } catch (NoSuchMethodException e1) {
+        throw new UnsupportedOperationException(
+            "Can't find member " + codeSignature.toLongString());
+      }
```

Comment:
@balalakshmifk : did you get a chance to check this?

Reply:
There were UTs for intercepting private methods. But it was working fine because the mock interceptor does not access `methodInvocation.getMethod()`, which internally calls this `getStaticParts()`. I have now added getMethod call to mock interceptor which covers this class under existing UTs.

Reply:
sure, can you confirm that with updated UT and without the above code fix (to access getDeclaredMethod()), UT fails correctly?

Reply:
Yes. It failed for updated UT with old code.